Sources/VLCMediaList.m

```diff
@@ -185,10 +191,9 @@ - (VLCMedia *)mediaAtIndex:(NSUInteger)index
     return media;
 }
 
-- (NSInteger)indexOfMedia:(VLCMedia *)media
+- (NSUInteger)indexOfMedia:(VLCMedia *)media
 {
-    NSInteger result = libvlc_media_list_index_of_item(p_mlist, [media libVLCMediaDescriptor]);
-    return result;
+    return [_mediaObjects indexOfObject:media];
```

Comment:
Do all these changes to indexOfMedia and removeMediaAtIndex work with discovered media in a network as well? Being in a WLAN with a NAS or twonky server for example should be enough to test these changes

Reply:
you can try it for 30 days :) https://twonky.com/

Reply:
and then you set it up build vlc ios with the custom vlckit and just check as it discovers files that all files  are discovered and that it doesn't crash somewhere unexpected and when you delete one out of the discovered folder that it all works as expected

Reply:
Just confirmed it still works and found [a bug](https://code.videolan.org/videolan/vlc-ios/issues/283) while at it :D